tree/tree/src/TBranch.cxx

```diff
@@ -1279,6 +1279,14 @@ Int_t TBranch::GetEntry(Long64_t entry, Int_t getall)
             fNextBasketEntry = -1;
             return -1;
          }
+         if (fTree->GetMaxVirtualSize() < 0) {
```

Comment:
On re-using GetMaxVirtualSize to enable the 'load-whole-cluster-at-once' feature.

In first approximation this seems to be two distincts features and I could imagine the user wanting both  'load-whole-cluster-at-once'  and 'keep more one basket per branch in memory'

I.e. TBranch::GetFreshBasket starts with testing TTree::MemoryFull, GetFreshCluster could also do that .... (we could use fBits to record the information).

Reply:
Hi Philippe,

I added a TTree::SetClusterPrefetch() to allow separate setting of the pre-fetch capability.

Peter